classes/OpeningHours/Module/Shortcode/IsOpen.php

```diff
@@ -81,12 +83,45 @@ public function shortcode ( array $attributes ) {
 
     $attributes['is_open'] = $isOpen;
     $attributes['classes'] .= ($isOpen) ? $attributes['open_class'] : $attributes['closed_class'];
-    $attributes['text'] = ($isOpen) ? $attributes['open_text'] : $attributes['closed_text'];
+
+    // If the attribute show_holiday is enabled
+    $closedText = $attributes['closed_text'];
+    if ($attributes['show_closed_holidays']) {
+        $holidaysList = $this->getTodaysHolidaysCommaSeperated($todayData);
+
+        $closedText = sprintf(
+            $holidaysList ? $attributes['closed_holiday_text'] : $closedText,
+            $holidaysList
+        );
+    }
+
+    $attributes['text'] = ($isOpen) ? $attributes['open_text'] : $closedText;
 
     echo $this->renderShortcodeTemplate($attributes, 'shortcode/is-open.php');
   }
 
   /**
+   * Retrieves holiday names for today
+   * @param  array $todayData   Data for today
+   * @return string            Extracted holiday name(s)
+   */
+  public function getTodaysHolidaysCommaSeperated($todayData) {
+      if (count($todayData['holidays']) > 0) {
+          $holidayNames = array();
+
+        foreach ($todayData['holidays'] as $holiday) {
+            array_push($holidayNames, $holiday->getName());
+        }
+
+        $names = implode(', ', $holidayNames);
+
+        return trim(rtrim($names, ','));
+    }
+
+    return false;
```

Comment:
This is not compliant with the method doc. Also I'd say that `null` would be better suited for this purpose.

Reply:
👍